tests/ut/test_get_vector.cc

```diff
@@ -17,25 +17,81 @@
 #include "knowhere/factory.h"
 #include "utils.h"
 
-TEST_CASE("Test Get Vector By Ids", "[GetVectorByIds]") {
+TEST_CASE("Test Binary Get Vector By Ids", "[Binary GetVectorByIds]") {
     using Catch::Approx;
 
     int64_t nb = 10000;
     int64_t dim = 128;
     int64_t seed = 42;
 
-    auto base_gen = [&]() {
+    auto base_bin_gen = [&]() {
         knowhere::Json json;
         json[knowhere::meta::DIM] = dim;
-        json[knowhere::meta::METRIC_TYPE] = knowhere::metric::L2;
+        json[knowhere::meta::METRIC_TYPE] = knowhere::metric::HAMMING;
         json[knowhere::meta::TOPK] = 1;
         return json;
     };
 
-    auto base_bin_gen = [&]() {
+    auto bin_ivfflat_gen = [&base_bin_gen]() {
+        knowhere::Json json = base_bin_gen();
+        json[knowhere::indexparam::NLIST] = 16;
+        json[knowhere::indexparam::NPROBE] = 4;
+        return json;
+    };
+
+    auto bin_flat_gen = base_bin_gen;
+
+    SECTION("Test binary index") {
+        using std::make_tuple;
+        auto [name, gen] = GENERATE_REF(table<std::string, std::function<knowhere::Json()>>({
+            make_tuple(knowhere::IndexEnum::INDEX_FAISS_BIN_IDMAP, bin_flat_gen),
+            make_tuple(knowhere::IndexEnum::INDEX_FAISS_BIN_IVFFLAT, bin_ivfflat_gen),
+        }));
+        auto idx = knowhere::IndexFactory::Instance().Create(name);
+        auto cfg_json = gen().dump();
+        CAPTURE(name, cfg_json);
+        knowhere::Json json = knowhere::Json::parse(cfg_json);
+        auto train_ds = GenBinDataSet(nb, dim, seed);
+        auto ids_ds = GenIdsDataSet(nb, dim);
+        REQUIRE(idx.Type() == name);
+        auto res = idx.Build(*train_ds, json);
+        REQUIRE(res == knowhere::Status::success);
+        knowhere::BinarySet bs;
+        idx.Serialize(bs);
+
+        auto idx_new = knowhere::IndexFactory::Instance().Create(name);
+        idx_new.Deserialize(bs);
+        auto results = idx_new.GetVectorByIds(*ids_ds, json);
+        REQUIRE(results.has_value());
+        auto xb = (uint8_t*)train_ds->GetTensor();
+        auto res_rows = results.value()->GetRows();
+        auto res_dim = results.value()->GetDim();
+        auto res_data = (uint8_t*)results.value()->GetTensor();
+        REQUIRE(res_rows == nb);
+        REQUIRE(res_dim == dim);
+        const auto data_bytes = dim / 8;
+        for (int i = 0; i < nb; ++i) {
+            auto id = ids_ds->GetIds()[i];
+            for (int j = 0; j < data_bytes; ++j) {
+                REQUIRE(res_data[i * data_bytes + j] == xb[id * data_bytes + j]);
+            }
+        }
+    }
+}
+
+TEST_CASE("Test Float Get Vector By Ids", "[Float GetVectorByIds]") {
+    using Catch::Approx;
+
+    int64_t nb = 10000;
+    int64_t dim = 128;
+    int64_t seed = 42;
+
+    auto metric = GENERATE(as<std::string>{}, knowhere::metric::L2, knowhere::metric::COSINE);
```

Comment:
how about IP?

Reply:
for float index, when COSINE metric type, get vector may return not-original vector.
so it's enough to only test L2 and COSINE